components/eventing-controller/controllers/subscriptionv2/jetstream/reconciler_internal_unit_test.go

```diff
@@ -202,28 +243,41 @@ func Test_handleSubscriptionDeletion(t *testing.T) {
 			err := r.Client.Create(testEnvironment.Context, sub)
 			require.NoError(t, err)
 
-			mockedBackend.On("DeleteSubscription", sub).Return(nil)
+			if testCase.wantError == errFailedToRemoveFinalizer {
+				sub.ObjectMeta.ResourceVersion = ""
+			}
+
+			if testCase.wantDeleteCall {
+				if testCase.wantError == errFailedToDeleteSub {
+					mockedBackend.On("DeleteSubscription", sub).Return(errors.New("deletion error"))
+				} else {
+					mockedBackend.On("DeleteSubscription", sub).Return(nil)
+				}
+			}
 
 			// when
 			err = r.handleSubscriptionDeletion(ctx, sub, r.namedLogger())
-			require.NoError(t, err)
+			require.ErrorIs(t, err, testCase.wantError)
 
 			// then
-			if testCase.wantDeleteCall {
-				mockedBackend.AssertCalled(t, "DeleteSubscription", sub)
+			mockedBackend.AssertExpectations(t)
+
+			if testCase.wantError == errFailedToRemoveFinalizer {
```

Comment:
(nit)

since this hack with `errFailedToRemoveFinalizer` is only used for one table tests, that could be a good indicator, that these table test could be extracted into a separate unit tests.

In that case you would get rid of the if-else clauses and of the setting the finalizers to nil in the end of the test.

Reply:
I would leave this as is since it's a trade off both ways.